atspi-macros/src/utils.rs

```diff
@@ -0,0 +1,242 @@
+use std::collections::HashMap;
+
+use proc_macro2::TokenStream;
+use proc_macro_crate::{crate_name, FoundCrate};
+use quote::{format_ident, quote};
+use syn::{
+    Attribute, FnArg, Ident, Lit, Meta, MetaList, NestedMeta, Pat, PatIdent, PatType, Result,
+};
+
+pub fn zbus_path() -> TokenStream {
+    if let Ok(FoundCrate::Name(name)) = crate_name("zbus") {
+        let ident = format_ident!("{}", name);
+        quote! { ::#ident }
+    } else {
+        quote! { ::zbus }
+    }
+}
+
+pub fn typed_arg(arg: &FnArg) -> Option<&PatType> {
+    match arg {
+        FnArg::Typed(t) => Some(t),
+        _ => None,
+    }
+}
+
+pub fn pat_ident(pat: &PatType) -> Option<&Ident> {
+    match &*pat.pat {
+        Pat::Ident(PatIdent { ident, .. }) => Some(ident),
+        _ => None,
+    }
+}
+
+// Convert to pascal case, assuming snake case.
+// If `s` is already in pascal case, should yield the same result.
+pub fn pascal_case(s: &str) -> String {
```

Comment:
Nitpick: [UpperCamelCase](https://github.com/rust-lang/rfcs/blob/master/text/0430-finalizing-naming-conventions.md) is more common in Rust than PascalCase.